packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -198,6 +198,142 @@ class RenderEditable extends RenderBox {
 
   Rect _lastCaretRect;
 
+  bool _rightArrowDown = false;
+  bool _leftArrowDown = false;
+  bool _upArrowDown = false;
+  bool _downArrowDown = false;
+
+  static const int _kLeftArrowCode = 21;
+  static const int _kRightArrowCode = 22;
+  static const int _kUpArrowCode = 19;
+  static const int _kDownArrowCode = 20;
+
+  int _extentOffset = -1;
+  int _baseOffset = -1;
+
+  int _previousCursorLocation = -1;
+  bool _resetCursor = false;
+
+  static const int _kShiftMask = 1;
+  static const int _kControlMask = 1 << 12;
+
+  void _handleKeyEvent(RawKeyEvent keyEvent){
+    if(defaultTargetPlatform != TargetPlatform.android)
```

Comment:
nit: space after if

Reply:
though maybe this would be better as a switch, where the iOS branch says that this is handled by the ObjectiveC code and the Fuchsia branch has a TODO?